editor/src/clj/editor/console.clj

```diff
@@ -336,87 +337,7 @@
-  (run [view-node] (view/move! view-node :selection :right)))
-
-(handler/defhandler :prev-word :console-view
-  (run [view-node] (view/move! view-node :navigation :prev-word)))
-
-(handler/defhandler :select-prev-word :console-view
-  (run [view-node] (view/move! view-node :selection :prev-word)))
-
-(handler/defhandler :next-word :console-view
-  (run [view-node] (view/move! view-node :navigation :next-word)))
-
-(handler/defhandler :select-next-word :console-view
-  (run [view-node] (view/move! view-node :selection :next-word)))
-
-(handler/defhandler :beginning-of-line :console-view
-  (run [view-node] (view/move! view-node :navigation :line-start)))
-
-(handler/defhandler :select-beginning-of-line :console-view
-  (run [view-node] (view/move! view-node :selection :line-start)))
-
-(handler/defhandler :beginning-of-line-text :console-view
-  (run [view-node] (view/move! view-node :navigation :home)))
-
-(handler/defhandler :select-beginning-of-line-text :console-view
-  (run [view-node] (view/move! view-node :selection :home)))
-
-(handler/defhandler :end-of-line :console-view
-  (run [view-node] (view/move! view-node :navigation :end)))
-
-(handler/defhandler :select-end-of-line :console-view
-  (run [view-node] (view/move! view-node :selection :end)))
-
-(handler/defhandler :page-up :console-view
-  (run [view-node] (view/page-up! view-node :navigation)))
-
-(handler/defhandler :select-page-up :console-view
-  (run [view-node] (view/page-up! view-node :selection)))
-
-(handler/defhandler :page-down :console-view
-  (run [view-node] (view/page-down! view-node :navigation)))
-
-(handler/defhandler :select-page-down :console-view
-  (run [view-node] (view/page-down! view-node :selection)))
-
-(handler/defhandler :beginning-of-file :console-view
-  (run [view-node] (view/move! view-node :navigation :file-start)))
-
-(handler/defhandler :select-beginning-of-file :console-view
-  (run [view-node] (view/move! view-node :selection :file-start)))
-
-(handler/defhandler :end-of-file :console-view
-  (run [view-node] (view/move! view-node :navigation :file-end)))
-
-(handler/defhandler :select-end-of-file :console-view
-  (run [view-node] (view/move! view-node :selection :file-end)))
-
-(handler/defhandler :copy :console-view
-  (enabled? [view-node evaluation-context] (view/has-selection? view-node evaluation-context))
-  (run [view-node clipboard] (view/copy! view-node clipboard)))
-
-(handler/defhandler :select-all :console-view
-  (run [view-node] (view/select-all! view-node)))
-
-(handler/defhandler :select-next-occurrence :console-view
-  (run [view-node] (view/select-next-occurrence! view-node)))
-
-(handler/defhandler :select-next-occurrence :console-tool-bar
-  (run [view-node] (view/select-next-occurrence! view-node)))
-
-(handler/defhandler :split-selection-into-lines :console-view
-  (run [view-node] (view/split-selection-into-lines! view-node)))
+(view/register-fundamental-read-only-handlers! *ns* :console-view :console-tool-bar)
```

Comment:
We now call this shared function to register the "read-only" code view handlers with the Console view instead of duplicating the `defhandler` clauses across both files.